test/remote/gateways/remote_datatrans_test.rb

```diff
@@ -0,0 +1,187 @@
+    @bad_amount = 100000 # anything grather than 500 EUR
+
+    @options = {
+      order_id: SecureRandom.random_number(1000000000).to_s,
+      description: 'An authorize',
+      email: 'john.smith@test.com'
+    }
+
+    @billing_address = address
+
+    @execute_threed = {
+      execute_threed: true,
+      redirect_url: 'http://www.example.com/redirect',
+      callback_url: 'http://www.example.com/callback',
+      three_ds_2: {
+        browser_info:  {
+          width: 390,
+          height: 400,
+          depth: 24,
+          timezone: 300,
+          user_agent: 'Spreedly Agent',
+          java: false,
+          javascript: true,
+          language: 'en-US',
+          browser_size: '05',
+          accept_header: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
+        }
+      }
+    }
+  end
+
+  def test_successful_authorize
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+  end
+
+  def test_failed_authorize
+    # the bad amount currently is only setle to EUR currency
+    response = @gateway.purchase(@bad_amount, @credit_card, @options.merge({ currency: 'EUR' }))
+    assert_failure response
+    assert_equal response.error_code, 'BLOCKED_CARD'
+    assert_equal response.message, 'card blocked'
+  end
+
+  def test_failed_authorize_invalid_currency
+    response = @gateway.purchase(@amount, @credit_card, @options.merge({ currency: 'DKK' }))
+    assert_failure response
+    assert_equal response.error_code, 'INVALID_PROPERTY'
+    assert_equal response.message, 'authorize.currency'
+  end
+
+  def test_successful_capture
+    authorize_response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorize_response
+
+    response = @gateway.capture(@amount, authorize_response.authorization, @options)
+    assert_success response
+    assert_equal authorize_response.authorization, response.authorization
+  end
+
+  def test_successful_refund
+    purchase_response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success purchase_response
+
+    response = @gateway.refund(@amount, purchase_response.authorization, @options)
+    assert_success response
+  end
+
+  def test_successful_capture_with_less_authorized_amount_and_refund
+    authorize_response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success authorize_response
+
+    capture_response = @gateway.capture(@amount - 100, authorize_response.authorization, @options)
+    assert_success capture_response
+    assert_equal authorize_response.authorization, capture_response.authorization
+
+    response = @gateway.refund(@amount - 200, capture_response.authorization, @options)
```

Comment:
💬 Comment:

Ahhhhhhh now I can see why that weird authorization para on capture and void, you are getting this wrong, see => the reference transaction in this case is ALWAYS the authorize/purchase if you are going to do 1 or more captures/voids the merchant is always going to use the initial transaction as reference.

Reply:
They don't provide any value,  just a 204 code.

Reply:
I got it,